tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/se/SearchEngineUserReader.java

```diff
@@ -36,19 +39,14 @@ public Optional<UserDTO> getCurrentUserBy(final Authentication authentication) {
     if (authentication == null) {
       return Optional.empty();
     }
-    final Object principal = authentication.getPrincipal();
-    if (principal instanceof CamundaUser) {
-      final CamundaUser user = (CamundaUser) principal;
+    if (authentication.getPrincipal() instanceof final CamundaPrincipal principal) {
```

Comment:
❓ I'm not sure about this change in this class for Tasklist, this class is used when the auth is nor SSO or Identity (management Identity) and is not bound by the `consolidated-auth` profile.

I think it would be clearer if we mirrored the changes from Operate and had a `@Profile("consoludated-auth")` implementation of `UserReader`. This way we won't unintentionally delete the class down the line (these legacy classes will be removed in due course)